README.md

```diff
@@ -33,23 +33,158 @@ Main features
 - :microscope: Tiny and tree-shakeable
 - :muscle: Supports TypeScript
 
-## Docs
+## Table of Contents
+
+- [Installation](#installation)
+- [API](#api)
+  - [connectObservable](#connectObservable)
+  - [connectFactoryObservable](#connectFactoryObservable)
+  - [distinctShareReplay](#distinctShareReplay)
+  - [SUSPENSE](#SUSPENSE)
+  - [suspend](#suspend)
+  - [suspended](#suspended)
+  - [switchMapSuspended](#switchMapSuspended)
+  - [createInput](#createInput)
+- [Examples](#examples)
+
+
+## Installation
+    npm install re-rxjs
+
+## API
+
+### connectObservable
+```ts
+const [useCounter, sharedCounter$] = connectObservable(
+  clicks$.pipe(
+    scan(prev => prev + 1, 0),
+    startWith(0),
+  )
+)
+```
+Returns a hook that provides the latest update of the observable, and an enhanced 
+version of the original observable (the one that the hook will use), which is shared 
+and "replayable".
+
+### connectFactoryObservable
+```tsx
+const [useStory, getStory$] = connectFactoryObservable(
+  (storyId: number) => getStoryWithUpdates$(storyId)
+)
+
+const Story: React.FC<{id: number}> = ({id}) => {
+  const story = useStory(id);
+
+  return (
+    <article>
+      <h1>{story.title}</h1>
+      <p>{story.description</p>
+    </article>
+  )
+}
+```
+Returns a hook that receives as arguments the same arguments as the factory function 
+and provides the latest update of the observable returned by the factory function.
+
+It also returns a function that provides the shared observable that the hook uses, 
+which is shared and "replayable". That observable is disposed when its `refCount`
+goes down to zero.
+
+### distinctShareReplay
+
+A RxJS pipeable operator which performs a custom `shareReplay` that can be very useful 
```

Comment:
Lose "very"

Should we describe why it is useful? 
1: It ensures React components only update when necessary 
2: Components using the same hooks use the same underlying Observables
3: Unsubscribes to the underlying observable when all components using a hook are unmounted.

Though this might be too much. Just an Idea.